### PR TITLE
fix(aws-cloudfront): Add sslSupportMethod

### LINKED ISSUE #19476: (aws-cloudfront): sslSupportMethod is not defined
### What is the problem?

Cannot specify `sslSupportMethod` for `CloudFront Distribution` in `distribution.ts`.
I can specify `minimumProtocolVersion` but not `sslSupportMethod`.

Example of `minimumProtocolVersion`
https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cloudfront/lib/distribution.ts#L222
https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cloudfront/lib/distribution.ts#L537

Example of `sslSupportMethod`
https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cloudfront/lib/distribution.ts#L528

### Reproduction Steps

```ts
minimumProtocolVersion: cloudfront.SecurityPolicyProtocol.TLS_V1_2_2021,
sslSupportMethod: cloudfront.SSLMethod.SNI,
```

### What did you expect to happen?

I want to set `sslSupportMethod`.

### What actually happened?

I can't configure it with the new method `CloudFront Distribution`.
It seems to be possible with the old method `CloudFrontWebDistribution`.

### CDK CLI Version

v2.15.0

### Framework Version

v2.15.0

### Node.js Version

v16.10.0

### OS

M1 Mac (macOS Monterey v12.2.1)

### Language

Typescript

### Language Version

v3.9.7

### Other information

_No response_

### PR DESCRIPTION
**Pull request by dayjournal**
- Modified to allow selection of sslSupportMethod

closes #19476

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [ ] Did you use `cdk-integ` to deploy the infrastructure and generate the snapshot (i.e. `cdk-integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cloudfront/lib/distribution.ts [hunk 1]
```diff
@@ -220,6 +220,25 @@ export interface DistributionProps {
     * @default - SecurityPolicyProtocol.TLS_V1_2_2021 if the '@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021' feature flag is set; otherwise, SecurityPolicyProtocol.TLS_V1_2_2019.
     */
   readonly minimumProtocolVersion?: SecurityPolicyProtocol;
+
+  /**
+    * The SSL method CloudFront will use for your distribution.
+    *
+    * Server Name Indication (SNI) - is an extension to the TLS computer networking protocol by which a client indicates
+    *  which hostname it is attempting to connect to at the start of the handshaking process. This allows a server to present
+    *  multiple certificates on the same IP address and TCP port number and hence allows multiple secure (HTTPS) websites
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-cloudfront/lib/distribution.ts:229]: can we remove the extra space at the front of these two lines?

File: packages/@aws-cdk/aws-cloudfront/lib/distribution.ts [hunk 2]
```diff
@@ -220,6 +220,25 @@ export interface DistributionProps {
...
+    * (or any other service over TLS) to be served by the same IP address without requiring all those sites to use the same certificate.
+    *
+    * CloudFront can use SNI to host multiple distributions on the same IP - which a large majority of clients will support.
+    *
+    * If your clients cannot support SNI however - CloudFront can use dedicated IPs for your distribution - but there is a prorated monthly charge for
+    * using this feature. By default, we use SNI - but you can optionally enable dedicated IPs (VIP).
+    *
+    * See the CloudFront SSL for more details about pricing : https://aws.amazon.com/cloudfront/custom-ssl-domains/
+    *
+    * @default - SSLMethod.SNI
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-cloudfront/lib/distribution.ts:239]: should be 

```
@default SSLMethod.SNI
```

**Review by comcalvi — DISMISSED** [review]
Looks great! Just a few small comments. Could we also add a test to this? At least check that we correctly fall back to SNI if the user doesn't specify an SSLMethod, and that if the user does specify an SSLMethod, we use the one they specified.

File: packages/@aws-cdk/aws-cloudfront/lib/distribution.ts (hunk 2 above)
Comment (dayjournal) [reply packages/@aws-cdk/aws-cloudfront/lib/distribution.ts]: Corrected comment.

**Comment by dayjournal** [comment]
@comcalvi : Hi, I modified the comment and added a test. Please check. 🙇

**Review by comcalvi — APPROVED** [review]
Thanks for the contribution

**Pull request merged**